news/12603.feature.rst

```diff
@@ -0,0 +1 @@
+When PEP-658 metadata is available, full distribution download no longer occurs when using dry-run mode on install.
```

Comment:
This needs to be updated slightly, in-particular all sdists are still downloaded (at least for now), either specifically call out wheels or change to something like:

> When PEP-658 metadata is available for a distribution the full file is no longer downloaded

Or

> When PEP-658 metadata is available, full wheel download no longer occurs

Further, also add that `pip lock` no longer downloads the distribution as well.

Reply:
If there's no update and no objections in the next few days I will update this news items myself.